language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/DamlDataTypeGen.scala

```diff
@@ -134,11 +135,56 @@ object DamlRecordOrVariantTypeGen {
           $valueInstanceExpr"""
     }
 
+    /**
+      *  The generated class for a DAML enum type contains:
+      *  - the definition of a "Value" trait
+      *  - the definition of a _case object_ for each constructor of the DAML enum
+      *  - A type class instance (i.e. implicit object) for serializing/deserializing
+      *    to/from the ArgumentValue type (see typed-ledger-api project)
+      */
+    def toScalaDamlEnumType(constructors: List[Ref.Name]): (Set[Tree], (Tree, Tree)) = {
+      val className = damlScalaName.name.capitalize
+
+      val klass =
+        q"""
+          sealed abstract class ${TypeName(className)}(
+            override val constructor: String,
+            override val index: Int
+          ) extends ${tq"$domainApiAlias.EnumRef"} {
+            ..$rootClassChildren
+          }"""
+
+      val (imports, companionObject) = constructors match {
+        case firstValue :: otherValues =>
+          Set(LFUtil.domainApiImport) ->
+            q"""
+            object ${TermName(className)} extends
+              ${tq"$domainApiAlias.EnumCompanion[$appliedValueType]"} {
+              ..${constructors.zipWithIndex.map {
+              case (c, i) =>
+                q"""case object ${TermName(c.capitalize)} extends $appliedValueType($c, $i) """
+            }}
+            
+              val firstValue: $appliedValueType = ${TermName(firstValue.capitalize)}
+              val otherValues:  Vector[$appliedValueType] =
```

Comment:
Add `stdVectorType` alongside `stdMapType` to `LFUtil`, and

```suggestion
              val otherValues:  $stdVectorType[$appliedValueType] =
```

Reply:
fixed